sync/class.jetpack-sync-settings.php

```diff
@@ -93,11 +109,19 @@ static function get_setting( $setting ) {
 	static function update_settings( $new_settings ) {
 		$validated_settings = array_intersect_key( $new_settings, self::$valid_settings );
 		foreach ( $validated_settings as $setting => $value ) {
-			update_option( self::SETTINGS_OPTION_PREFIX . $setting, $value, true );
+
+			if ( self::is_network_setting( $setting ) ) {
+				if ( is_multisite() && is_main_site() ) {
+					update_site_option( self::SETTINGS_OPTION_PREFIX . $setting, $value );
+				}
+			} else {
+				update_option( self::SETTINGS_OPTION_PREFIX . $setting, $value, true );
+			}
+
 			unset( self::$settings_cache[ $setting ] );
 
 			// if we set the disabled option to true, clear the queues
-			if ( 'disable' === $setting && ! ! $value ) {
+			if ( ( 'disable' === $setting || 'network_disable' ) && ! ! $value ) {
```

Comment:
Are we missing a ` === $setting` after `|| 'network_disable'` here?
```suggestion
			if ( ( 'disable' === $setting || 'network_disable' === $setting ) && ! ! $value ) {
```